Fixes issue #124 JSON is overwritten if unable to parse the file in DEV

Netgator opened the pull request:
Fixes issue #124 JSON is overwritten if unable to parse the file in DEV mode.

If locale JSON file exists AND package is in DEV mode AND the locale is undefined (unreadable) - This HALTS ALL execution by throwing an exception which prevents the locale file(s) from being overwritten inadvertently.  The issue is when the JSON is invalid (even just missing a comma) the file is automatically overwritten which can cause developers to loose most existing lines from the file and no way to recover...

Netgator committed:
Fixes issue #124 JSON is overwritten if unable to parse the file in DEV mode.

If JSON file exists and in DEV mode but the locale is undefined - HALTS ALL execution with an expection to prevent the files from being overwritten.

gjuchault commented on i18n.js:
```diff
@@ -330,6 +330,10 @@ i18n.prototype = {
 
 			locale = this.defaultLocale;
 
+			if ((this.devMode) && (fs.existsSync(this.locateFile(locale)))) {
```
Why would this condition would match if the defaultLocale have parsing errors ?

gjuchault replied:
I think you want this to be over line 331

Netgator replied:
This code matches (and throws an error) when json is present BUT is (for example) missing one comma...   You can test by creating a file with two comments and then remove the comma from the en.json file. Without this code the file is rewritten when a parse error occurs which potentially erases thousands of lines due to a missing comma :-)

This code prevents a json file from being overwritten when it has content but can not be parsed correctly.

Netgator replied:
Don't think we want the extra line because the problem would occur in ANY language not just the default language...

Netgator committed:
updates to make eslint compatible

updated code to make it compatiable with most current air-bnb /QSS eslint rules.
researched syncronis functions and realized that this only applies in DEFV mode - otherwise the files are cached on startup.

Netgator committed:
Implemented FIX #94 from Jeremylevy - Dec 2016

Explicit language parameter in query, domain or session must have precedence over "preferred" language present in the "Accept-Language" header.

Netgator committed:
Implemented Query Fix submitted by tpgmartin - Aug 2015

Optionally pass custom query string parameter

Netgator committed:
corrected default extension to be .json

update to make the default extension to be .json rather than .js

Netgator committed:
update to make production the default unless dev is specified...

Netgator committed:
remove guilt from build process

artijoan committed:
Update i18n.js

Netgator committed:
Merge pull request #1 from artijoan/master

Update i18n.js tonprevent crashes

jegarcia2 committed:
Dependency updates for NODE 22

jegarcia2 committed:
Airbnb base - Hotfix

jegarcia2 committed:
NODE 22 - Package json update

Netgator committed:
Merge pull request #2 from jegarcia2/NODE-22-UPDATE

Dependency updates for NODE 22
